src/ucs/arch/cpu.c

```diff
@@ -0,0 +1,106 @@
+/**
+* Copyright (C) Mellanox Technologies Ltd. 2019. ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#ifdef HAVE_CONFIG_H
+#  include "config.h"
+#endif
+
+#include <ucs/arch/cpu.h>
+#include <ucs/arch/generic/cpu.h>
+#include <ucs/sys/sys.h>
+#include <ucs/sys/string.h>
+#include <ucs/sys/stubs.h>
+#include <ucs/type/init_once.h>
+
+#define UCS_CPU_CACHE_FILE_FMT     "/sys/devices/system/cpu/cpu%d/cache/index%d/%s"
+#define UCS_CPU_CACHE_LEVEL        "level"
+#define UCS_CPU_CACHE_TYPE         "type"
+#define UCS_CPU_CACHE_SIZE         "size"
+
+/* cache size array. index - cache type (ucs_cpu_cache_type_t), value - cache value,
+ * 0 means cache is not supported */
+static size_t ucs_cpu_cache_size[UCS_CPU_CACHE_LAST] = {0};
+
+static ucs_init_once_t ucs_cache_read_once = UCS_INIT_ONCE_INIITIALIZER;
+
+/* cache datatypes */
+struct { /* cache info obtained from [linux] kernel */
+    int             level;
+    const char     *type;
+} const ucs_cpu_cache_sysfs_name[] = {
+    [UCS_CPU_CACHE_L1d] = {.level = 1, .type = "Data"},
+    [UCS_CPU_CACHE_L1i] = {.level = 1, .type = "Instruction"},
+    [UCS_CPU_CACHE_L2]  = {.level = 2, .type = "Unified"},
+    [UCS_CPU_CACHE_L3]  = {.level = 3, .type = "Unified"}
+};
+
+
+static void ucs_sysfs_get_cache_size(size_t *cache, size_t length)
```

Comment:
maybe no need to pass cache and length?
assume it's ucs_cpu_cache_size and UCS_CPU_CACHE_LAST